### DISCUSSION TITLE
Authenticate user in Webhook

### DISCUSSION THREAD

**Discussion by sushilsth** [osiset/laravel-shopify#1135] [Q&A]
Hi, I am new to shopify app development. I want to make ADMIN API call in Laravel Jobs that is when webhook notification is triggered.

Here is the details of app flow:
1. Customer creates order in merchant shop
2. Order-create webhook is triggered
3. When order-create webhook notification is received I want to call billing api with type usage charge.

The main problem I have is how to authenticate user in webhook. I know to authenticate the user within Shopify admin using `$shop = Auth::user();  $shopApi = $shop->api()->rest('GET', '/admin/shop.json')['body']['shop'];`. But this donot work in Laravel Job handle method.

I was thinking to save the access-token that is generated " $shop = Auth::user(); $shopApi = $shop->api()". But I can't seem to find the way to access the access_token and save it into the database so that I can use same access_token to make billing api requests.
![image](https://user-images.githubusercontent.com/7147132/168540849-46015730-b5a3-4a7e-82c9-736b3af97c78.png)

Is this approach the best approach. Does access_token has any expiry date?

Please recommend if there is any best approach as well.

Thank you

**Comment by khaiknievel** [comment]
To get the $shop, you can refer to this [file.](https://github.com/osiset/laravel-shopify/blob/8eb018117086ea5f0b311b018766749e67d2af8e/src/Messaging/Jobs/AppUninstalledJob.php#L73)
